pages/studentAssessmentExam/studentAssessmentExam.js

```diff
@@ -1,201 +1,214 @@
-          permissions,
-          minsize,
-          maxsize,
-          groupsize,
-          needsize,
-          group_info,
-          join_code,
-          start,
-          used_join_code
-        ) {
-          if (ERR(err, next)) return;
-          res.locals.permissions = permissions;
-          res.locals.minsize = minsize;
-          res.locals.maxsize = maxsize;
-          res.locals.groupsize = groupsize;
-          res.locals.needsize = needsize;
-          res.locals.group_info = group_info;
-          res.locals.join_code = join_code;
-          res.locals.start = start;
-          res.locals.used_join_code = used_join_code;
-          res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+router.get(
+  '/',
+  asyncHandler(async function (req, res, next) {
+    if (res.locals.assessment.type !== 'Exam') return next();
+    var params = {
+      assessment_id: res.locals.assessment.id,
+      user_id: res.locals.user.user_id,
+    };
+    if (res.locals.assessment.multiple_instance) {
+      // The user has landed on this page to create a new assessment instance.
+      //
+      // Before allowing the user to create a new assessment instance, we need
+      // to check if the current access rules require a password. If they do,
+      // we'll ensure that the password has already been entered before allowing
+      // students to create and start a new assessment instance.
+      if (!checkPasswordOrRedirect(req, res)) return;
+      if (res.locals.assessment.group_work) {
+        // Get the group config info
+        const groupConfig = await groupAssessmentHelper.getGroupConfig(res.locals.assessment.id);
+        res.locals.groupConfig = groupConfig;
+
+        // Check whether the user is currently in a group in the current assessment by trying to get a group_id
+        const groupId = await groupAssessmentHelper.getGroupId(
+          res.locals.assessment.id,
+          res.locals.user.user_id
+        );
+
+        if (groupId === null) {
+          res.locals.notInGroup = true;
+        } else {
+          const groupInfo = await groupAssessmentHelper.getGroupInfo(groupId, groupConfig);
+          res.locals.groupSize = groupInfo.groupSize;
+          res.locals.groupMembers = groupInfo.groupMembers;
+          res.locals.joinCode = groupInfo.joinCode;
+          res.locals.groupName = groupInfo.groupName;
+          res.locals.start = groupInfo.start;
+          res.locals.rolesInfo = groupInfo.rolesInfo;
+
+          if (groupConfig.hasRoles) {
+            const result = await groupAssessmentHelper.getAssessmentLevelPermissions(
+              res.locals.assessment.id,
+              res.locals.user.user_id
+            );
+            res.locals.canViewRoleTable = result.can_assign_roles_at_start;
+          }
         }
-      );
+
+        res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+      } else {
+        res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+      }
```

Comment:
Nit, minor simplification:

```suggestion
      }
      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
```